### DISCUSSION TITLE
git LFS does not find github ?

### DISCUSSION THREAD

**Discussion by mskutnik** [git-lfs/git-lfs#5218] [General]
i was downloading this for the first time, and the only message I get from the git installer is just a "Could not find Git; can not register to Git LFS." . Has this happened to anyone else ?

**Comment by bk2204** [comment]
Hey,

Usually the best way to install Git LFS on Windows is as part of Git for Windows.  Git LFS requires Git to be installed, so if you don't have Git installed, then it won't work.  If you do have it installed, then it needs to be in the `PATH` so Git LFS can find it.

You can see how @chrisd8088 explained this pretty well in #5174.

**Comment by chrisd8088** [comment]
This sound like you don't have Git installed or it is not accessible in your `PATH` to the Git LFS extension.

If you are using the [Git for Windows](https://gitforwindows.org/) project's installer and have chosen the **"Use Git from Git Bash only"** option, that can, for now, be the cause this problem.

See #5031 for some discussion of this problem and an explanation of why you need to choose another option for now, such as **"Git from the command line and also from 3rd-party software"**.

There will be a fix for this particular problem in the next Git for Windows release, due next week; that will package Git version 2.39 and Git LFS version 3.3.0, which includes the changes from PR #5048.

It would be appreciated if you can try this Git for Windows pre-release installer and see if it fixes your problem!

https://github.com/git-for-windows/git/releases/tag/v2.39.0-rc2.windows.1

---

Also note that Git, Git LFS, Git for Windows, and GitHub are all separate things.

- Git is the source code management tool.
- Git LFS is an extension to Git to support large files.
- Git for Windows is a project which packages Git, Git LFS, and other tools specifically for Windows, and includes an Windows installer for these.
- GitHub is a commercial provider of a Git repository hosting service, as are other providers like GitLab.

**Comment by TG-009** [comment]
I have installed Git for windows and Git LFS and I am still having issues making git LFS work.  Maybe my issue has something to do with the PATH.
It is mentioned that "...or it is not accessible in your PATH to the Git LFS extension."
Can someone please explain this for me
How do I know if it (whatever it is) is or is not accessible in the PATH???
Where do I set this path and where do I check the Path??

**Comment by chrisd8088** [comment]
First of all, please try the latest Git for Windows release (https://github.com/git-for-windows/git/releases/tag/v2.39.0.windows.2), as I think it may resolve your specific problem.

Second, the `PATH` environment variable is described in various sources such as this Wikipedia [article](https://en.wikipedia.org/wiki/PATH_(variable)#DOS,_OS/2,_and_Windows); it contains a list of filesystem locations which should be searched in the given order to find executable files.

For instance, both Git and Git LFS need to execute each other, so when `git.exe` is running it needs to find and run `git-lfs.exe`, and when `git-lfs.exe` is running it needs to find and run `git.exe`.  To do this, the list of filesystem locations in the `PATH` environment variable are searched until an executable program file is found that matches the requested program's name.  If no matching file found, you get an error.

The Git for Windows installer tries to set up the `PATH` variable with an appropriate set of filesystem locations so this all works transparently, but prior to v2.39.0, if you choose **"Use Git from Git Bash only"** then this doesn't work properly.

There are lots of resources online regarding how one can set the `PATH` variable manually, either through the Windows UI, in a `cmd` shell session, or in a PowerShell session.  But you should definitely try the Git for Windows installer first.

**Reply by TG-009** [reply]
Thank you for the clarification,  I was under the assumption that the PATH discussion was referring to something within a git-lfs file somewhere.
I did install git for windows v2.39.0 and DID NOT choose "Use Git from Git Bash only".
